client/components/Admin/CreateNewUser.tsx

```diff
@@ -0,0 +1,213 @@
+import React, { FC, useEffect, useState } from "react";
+import { Flex } from "reflexbox/styled-components";
+import Text, { H2 } from "../Text";
+import { Button } from "../Button";
+import Modal from "../Modal";
+import styled from "styled-components";
+import { fadeIn } from "../../helpers/animations";
+import { useStoreState } from "../../store";
+import { useFormState } from "react-use-form-state";
+import Router from "next/router";
+import emailValidator from "email-validator";
+import axios from "axios";
+import { APIv2 } from "../../consts";
+import { getAxiosConfig } from "../../utils";
+import { Col, ColCenterV } from "../Layout";
+import Icon from "../Icon";
+import { TextInput } from "../Input";
+
+const LoginForm = styled(Flex).attrs({
+  as: "form",
+  flexDirection: "column"
+})`
+  animation: ${fadeIn} 0.8s ease-out;
+`;
+
+const Email = styled.span`
+  font-weight: normal;
+  color: #512da8;
+  border-bottom: 1px dotted #999;
+`;
+
+interface CreateNewUserButtonProps {
+  reload: () => void;
+}
+
+const CreateNewUserButton: FC<CreateNewUserButtonProps> = ({ reload }) => {
+  const [showCreateUser, setShowCreateUser] = useState(false);
+
+  const onClose = () => {
+    reload();
+    setShowCreateUser(false);
+  };
+
+  return (
+    <>
+      <Button
+        height={[32, 40]}
+        color="purple"
+        onClick={() => setShowCreateUser(true)}
+      >
+        New User
+      </Button>
+      <Modal id="create-new-user" show={showCreateUser} closeHandler={onClose}>
+        <CreateNewUser close={onClose} />
+      </Modal>
+    </>
+  );
+};
+
+interface CreateNewUserProps {
+  close: () => void;
+}
+
+const CreateNewUser: FC<CreateNewUserProps> = ({ close }) => {
+  const { isAuthenticated, isAdmin } = useStoreState(s => s.auth);
+  const [error, setError] = useState("");
+  const [verifying, setVerifying] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [formState, { email, password, label }] = useFormState<{
+    email: string;
+    password: string;
+  }>(null, { withIds: true });
+
+  const reset = () => {
+    formState.clear();
+    setVerifying(false);
+  };
+
+  useEffect(() => {
+    if (!isAuthenticated) Router.push("/");
+  }, [isAuthenticated]);
+
+  function onSubmit() {
+    return async e => {
+      e.preventDefault();
+      const { email, password } = formState.values;
+
+      if (loading) return null;
+
+      if (!email) {
+        return setError("Email address must not be empty.");
+      }
+
+      if (!emailValidator.validate(email)) {
+        return setError("Email address is not valid.");
+      }
+
+      if (password.trim().length < 8) {
```

Comment:
Dh Passwörter dürfen nicht mit Leerzeichen beginnen oder enden?

Reply:
Hab ich so vom originalen signup übernommen.
Zumindest zählen Leerzeichen nicht zur min-Länge
Schau mir an wie's dann ans Backend weitergegeben wird bzw in die DB gespeichert